src/frontend/js/components/DashboardItem/DashboardItemEnrollment.spec.tsx

```diff
@@ -0,0 +1,63 @@
+import { IntlProvider } from 'react-intl';
+import { DashboardItemEnrollment } from 'components/DashboardItem/DashboardItemEnrollment';
+import { render, screen } from '@testing-library/react';
+import { Enrollment } from 'types/Joanie';
+import { JoanieEnrollmentFactory } from 'utils/test/factories';
+
+describe('<DashboardItemEnrollment/>', () => {
+  const enrollment: Enrollment = JoanieEnrollmentFactory.seed(3).generate();
+  enrollment.course_run = {
+    ...enrollment.course_run,
+    end: '2039-10-01T01:23:37+00:00',
+    start: '2022-09-09T12:02:44+00:00',
+  };
+
+  const course = enrollment.course_run.course!;
```

Comment:
We prefer to use full random values in order to detect edge cases.
I recommend you to create an enrollment for each tests.

You are able to generate a course run with a given state with random values. e.g:

```js
 const courseRun: CourseRun = JoanieCourseRunFactory.extend({
        // - Course Run opened for enrollment but not yet started
        ...cr,
        enrollment_start: faker.date.past(0.25)().toISOString(),
        enrollment_end: faker.date.future(0.5)().toISOString(),
        start: faker.date.future(0.75)().toISOString(),
        end: faker.date.future(1.0)().toISOString(),
        }).generate();

const enrollment = JoanieEnrollmentFactory.extend({ course_run: courseRun }).generate()
```